experimental/GModule/src/Cohomology.jl

```diff
@@ -1662,31 +1662,30 @@ iszero(a) || (@show g, h, k, a ; return false)
   return true
 end
 
+# create a free module with type "compatible" with that of `M`
+Oscar.free_module(M::FinGenAbGroup, n::Int) = free_abelian_group(n)
+Oscar.free_module(M::AbstractAlgebra.FPModule, n::Int) = free_module(base_ring(M), n)
+Oscar.free_module(M::Oscar.ModuleFP, n::Int) = FreeMod(base_ring(M), n)
```

Comment:
I agree that I find these a bit confusing, but due to another reason than for @mjrodgers.
consider the following example (which is from another area of Oscar, but maybe easier to understand):
```julia
R, x = polynomial_ring(QQ, :x)
S, y = polynomial_ring(R, :y)
```
The way the second call works is that it creates a polynomial ring *over* `R`, so it builds upon the first input.
The methods introduced here would however easily lead to the analogy that
`polynomial_ring(R::PolyRing, s::Symbol) = polynomial_ring(base_ring(R), s)`, i.e. that `S` is a polynomial ring over the same base ring as `R`. This is more what e.g. `similar` does in my opinion